pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/manual.py

```diff
@@ -45,7 +49,13 @@ def _heading_tag(self, token: Token, tokens: Sequence[Token], i: int, options: O
         (tag, attrs) = super()._heading_tag(token, tokens, i, options, env)
         if self._title_seen:
             if token.tag == 'h1':
-                raise RuntimeError("only one title heading allowed", token)
+                assert token.map is not None
+                raise RuntimeError(
+                    "only one title heading (# [text...]) allowed per manual chapter "
+                    f"but found a second in lines [{token.map[0]}..{token.map[1]}]. "
```

Comment:
Might be even greater to print the lines in question but it is great improvement already.